### PR TITLE
Add search to remaining Places pages

### PR DESCRIPTION
**Pull request by ToxicBananaParty**

### CODE REVIEW

**Review by myk002 — COMMENTED** [review]
make sure the Places `sort` additions are represented in `docs/changelog.txt`!

File: plugins/lua/sort/places.lua [hunk 1]
```diff
@@ -111,6 +111,28 @@ local function get_location_search_key(zone)
     return table.concat(result, ' ')
 end
 
+local function get_stockpile_search_key(stockpile)
+    -- I put the stockpile_number in twice so you can search both with and without the octothorpe
+    -- (e.g. if your unnamed stockpile is listed as "Stockpile #16" you can search both "#16" or "16")
```
Comment (myk002) [inline plugins/lua/sort/places.lua:116]: is this necessary? I would have thought that a search key of `Stockpile #16` would get matched by search text of either `16` or `#16`

**Comment by ToxicBananaParty** [comment]
Will do. I also need to add searching by assigned unit to workshops first.

File: plugins/lua/sort/places.lua [hunk 2]
```diff
@@ -111,6 +111,28 @@ local function get_location_search_key(zone)
...
+end
+
+local function get_workshop_search_key(workshop)
+    return ('%s %s %s'):format(workshop.name, df.workshop_type.attrs[workshop.type].name or '', df.workshop_type[workshop.type])
+end
+
+local function get_farmplot_search_key(farmplot)
+    local result = {}
+
+    if #farmplot.name ~= 0 then table.insert(result, farmplot.name) else table.insert(result, 'Farm Plot') end
```
Comment (myk002) [inline plugins/lua/sort/places.lua:128]: I'd think you'd want the word "farm" in the key regardless of whether the farmplot has a specific name, no?

Comment (ToxicBananaParty) [reply plugins/lua/sort/places.lua:128]: No, imo. This represents the default name for farms being "Farm Plot" -- there's no point in searching by 'farm' in a Farmplots page, you already know everything in that page is a farm.

File: plugins/lua/sort/places.lua (hunk 1 above)
Comment (ToxicBananaParty) [reply plugins/lua/sort/places.lua:116]: You're right -- I must have forgotten to make an unnamed stockpile first before testing that the first time, or something.

File: plugins/lua/sort/places.lua (hunk 2 above)
Comment (myk002) [reply plugins/lua/sort/places.lua:136]: oh, of course. in my mind I was picturing the workshops page. current code looks good then

File: plugins/lua/sort/places.lua [hunk 3]
```diff
@@ -112,14 +112,22 @@ local function get_location_search_key(zone)
...
-    else return ('Stockpile #%s %s'):format(stockpile.stockpile_number, stockpile.stockpile_number) end
+    else return ('Stockpile #%s'):format(stockpile.stockpile_number) end
 end
 
 local function get_workshop_search_key(workshop)
-    return ('%s %s %s'):format(workshop.name, df.workshop_type.attrs[workshop.type].name or '', df.workshop_type[workshop.type])
+    local result = {}
+    for _, unit_id in ipairs(workshop.profile.permitted_workers) do
+        local unit = df.unit.find(unit_id)
+        if unit then table.insert(result, sortoverlay.get_unit_search_key(unit)) end
```
Comment (myk002) [inline plugins/lua/sort/places.lua:123]: note that this will include the units' professions, which may or may not be desirable here

Comment (myk002) [reply plugins/lua/sort/places.lua:123]: meh, it's probably ok. what if you want to see where you assigned your stoneworkers?

Comment (ToxicBananaParty) [reply plugins/lua/sort/places.lua:123]: It lists their profession on the page so I figure that's good, in the interest of "can search any text that appears on the page"

**Review by myk002 — APPROVED** [review]

**Pull request merged**